lab/src/Microsoft.ServiceModel.Syndication/src/SyndicationFeed.cs

```diff
@@ -31,13 +28,101 @@ public class SyndicationFeed : IExtensibleSyndicationObject
         private ExtensibleSyndicationObject _extensions = new ExtensibleSyndicationObject();
         private string _generator;
         private string _id;
+
         private Uri _imageUrl;
+        private TextSyndicationContent _imageTitle;
+        private Uri _imageLink;
+        
         private IEnumerable<SyndicationItem> _items;
         private string _language;
         private DateTimeOffset _lastUpdatedTime;
         private Collection<SyndicationLink> _links;
         private TextSyndicationContent _title;
 
+        //new fields for new optional RSS tags
+        private SyndicationLink _documentation;
+        private int _timeToLive;
+        private Collection<int> _skipHours;
+        private Collection<string> _skipDays;
+        private SyndicationTextInput _textInput;
+        //private Uri _iconUrl;
+        private string _iconImage;
+
+        public string IconImage
+        {
+            get
+            {
+                return _iconImage;
+            }
+            set
+            {
+                if (value == null)
+                    throw new ArgumentNullException(nameof(value));
+                _iconImage = value;
+            }
+        } 
+
+        public SyndicationTextInput TextInput
+        {
+            get
+            {
+                //if (_textInput == null)
+                //    _textInput = new SyndicationTextInput();
```

Comment:
When writing the Feed, the writer checks if the TextInput field is null to know if it must or not write that tag, If we keep that code, it will never be null.
Deleting the comments. 

---
In reply to: [126765075](https://github.com/dotnet/wcf/pull/2028#discussion_r126765075) [](ancestors = 126765075)